doc/source/ray-core/ray-dag.rst

```diff
@@ -0,0 +1,155 @@
+.. _ray-dag-guide:
+
+Building Computation Graphs with Ray DAG API
+============================================
+
+With ``ray.remote`` you have the flexibility of running an application where
+computation is executed remotely at runtime. For a ``ray.remote`` decorated
+class or function, you can also use ``.bind`` on the body to build a static
+computation graph.
+
+.. note::
+
+     Ray DAG is designed to be developer facing API where recommended use cases
+     are
+     1) Locally iterate and test your application authored by higher level libraries.
+     2) Build libraries on top of the Ray DAG APIs.
+     Please see `DAG Unification Design Doc <https://docs.google.com/document/d/16wRobgsfCvOXnsaQq9lwQMDnnFWzXaYicIQ6e65W5W0>`_ for details.
+
+
+When ``.bind()`` is called on a ``ray.remote`` decorated class or function, it will
+generate an intermediate representation (IR) node that act as backbone and
+building blocks of the DAG that is statically holding the computation graph
+together, where each IR node is resolved to value at execution time with
+respect to the topological order.
+
+The IR node can also be assigned to a variable and passed into other nodes as
+arguments.
+
+Ray DAG with functions
+----------------------
+
+The IR node generated by ``.bind()`` on a ``ray.remote`` decorated function is
+executed as a Ray Task upon execution which will be solved to the task output.
+
+This example shows how to build a chain of functions where each node can be
+executed as root node while iterating, or used as input args or kwargs of other
+functions to form more complex DAGs.
+
+.. tabbed:: Python
+
+    .. code-block:: python
+
+        # `ray start --head` has been run to launch a local cluster
+        import ray
+
+        @ray.remote
+        def func(src, inc=1):
+            return src + inc
+
+        a_ref = func.bind(1, inc=2)
+        assert ray.get(a_ref.execute()) == 3 # 1 + 2 = 3
```

Comment:
this is the first time `execute()` is introduced. It shouldn't be in code. Can you explain it in a paragraph above?

Reply:
that's a good point but i want to change this to `dag.run()` in next PR and introduce it formally in the doc later

Reply:
wait is there a discussion changing execute to run? regardless, adding a short sentence is still encouraged here because you can just do find and replace later.

Reply:
yeah we had that email thread in April discussing ray.run() as well as stacking ray & serve decorators. Let me add a few sentences here